client/src/leap/soledad/client/api.py

```diff
@@ -70,6 +71,26 @@
 """
 SOLEDAD_CERT = None
 
+# A whitelist of modules from where to collect plugins dynamically.
+# For the moment restricted to leap namespace, but the idea is that we can pass
+# other "trusted" modules as options to the initialization of soledad.
+PLUGGABLE_LEAP_MODULES = ('mail', 'keymanager')
+
+
+def collect_plugins(interface):
+    """
+    Traverse a whitelist of modules and collect all the plugins that implement
+    the passed interface.
+    """
+    plugins = []
+    for namespace in PLUGGABLE_LEAP_MODULES:
+        try:
+            module = __import__('leap.%s.plugins' % namespace, fromlist='.')
+            plugins = plugins + list(getPlugins(interface, module))
```

Comment:
also, although mentioned: maybe opening up the namespace optionally is a good solution:

leap.plugins.* can be declared a namespace, and anyone can implement and package theirs :) 
(although I wouldn't activate it by default by the reasons explained above)